File: smartlist/src/main/java/com/smartlist/module/question/questionstrategy/TextStrategy.java
package com.smartlist.module.question.questionstrategy;

import com.smartlist.module.question.QuestionResponseCodes;
import com.smartlist.module.question.dto.request.QuestionReqDTO;
import com.smartlist.module.question.interfaces.IQuestionStrategy;

public class TextStrategy implements IQuestionStrategy {

    private final QuestionReqDTO questionReqDto;

    public TextStrategy(final QuestionReqDTO questionReqDto){
        this.questionReqDto = questionReqDto;
    }

    @Override
    public void validate() {
        if (!this.questionReqDto.getAnswerOptions().isEmpty()) {
            throw QuestionResponseCodes.INVALID_ANSWER_OPTIONS_FOR_QUESTION_TYPE;
        }
    }
}

File: javascript/smartlist/smartlist-api-mysql-project/src/modules/candidate/candidate-question-strategy/text-strategy.ts
import { Injectable } from '@nestjs/common';
import { Question } from 'src/entities/question.entity';
import { QuestionResponseCodes } from 'src/modules/question/question.response.codes';
import { SubmitAnswerReqDto } from '../dto/request/submit-answer.req.dto';
import { BaseQuestionStrategy } from './base-question-strategy';

@Injectable()
export class TextStrategy extends BaseQuestionStrategy {
  constructor(submitAnswerReqDto: SubmitAnswerReqDto, question: Question) {
    super(submitAnswerReqDto, question);
  }

  async validate() {
    if (!this.submitAnswerReqDto.answer_text) {
      throw QuestionResponseCodes.INVALID_ANSWER_OPTION;
    }
  }

  score(): number {
    return 0;
  }
}
